package.json

```diff
@@ -6,21 +6,25 @@
   "description": "The Internet, peer reviewed.",
   "scripts": {
     "build": "gulp build",
-    "checkformatting": "prettier --check *.js h/static/scripts",
-    "format": "prettier --list-different --write *.js h/static/scripts",
+    "checkformatting": "prettier --check *.{js,ts,tsx} h/static/scripts",
+    "format": "prettier --list-different --write *.{js,ts,tsx} h/static/scripts",
     "lint": "eslint h/static/scripts",
     "test": "gulp test"
```

Comment:
To run TS, we'd add a script like:

```
"typecheck": "tsc --build h/static/scripts/tsconfig.json"
```

To make that command run successfully there are a few minor corrections needed. The most notable one is that there are some things in JSX that are slightly different from HTML, and TS can catch these for you.

```diff
diff --git a/h/static/scripts/create-group-form.tsx b/h/static/scripts/create-group-form.tsx
index 4393d90de..f858a8ca4 100644
--- a/h/static/scripts/create-group-form.tsx
+++ b/h/static/scripts/create-group-form.tsx
@@ -14,8 +14,8 @@ function CreateGroupForm() {
             type="text"
             id="name"
             class="form-input__input group-form__name-input has-label"
-            autofocus=""
-            autocomplete="off"
+            autoFocus
+            autoComplete="off"
             required
           />
           <span class="form-input__character-counter is-ready">0/25</span>
@@ -57,7 +57,7 @@ function CreateGroupForm() {
 }
 
 function init() {
-  render(<CreateGroupForm />, document.querySelector('#create-group-form'));
+  render(<CreateGroupForm />, document.querySelector('#create-group-form')!);
 }
 
 init();
diff --git a/h/static/scripts/tsconfig.json b/h/static/scripts/tsconfig.json
index 33fd708ba..2aadc5b43 100644
--- a/h/static/scripts/tsconfig.json
+++ b/h/static/scripts/tsconfig.json
@@ -12,7 +12,7 @@
     "target": "ES2020",
     "useUnknownInCatchVariables": false
   },
-  "include": ["**/*.js", "**/*.ts", "**/*.tsx"],
+  "include": ["**/*.ts", "**/*.tsx"],
   "exclude": [
     // Tests and test infrastructure
     "**/test/*.js",
diff --git a/package.json b/package.json
index 4a8fac0ce..224c038c4 100644
--- a/package.json
+++ b/package.json
@@ -9,7 +9,8 @@
     "checkformatting": "prettier --check *.{js,ts,tsx} h/static/scripts",
     "format": "prettier --list-different --write *.{js,ts,tsx} h/static/scripts",
     "lint": "eslint h/static/scripts",
-    "test": "gulp test"
+    "test": "gulp test",
+    "typecheck": "tsc --build h/static/scripts/tsconfig.json"
   },
   "dependencies": {
     "@babel/core": "^7.24.7",
diff --git a/yarn.lock b/yarn.lock
index 3848796d9..7ec5987e2 100644
--- a/yarn.lock
+++ b/yarn.lock
@@ -2705,12 +2705,12 @@ __metadata:
   linkType: hard
 
 "@types/vinyl@npm:^2.0.9":
-  version: 2.0.11
-  resolution: "@types/vinyl@npm:2.0.11"
+  version: 2.0.12
+  resolution: "@types/vinyl@npm:2.0.12"
   dependencies:
     "@types/expect": ^1.20.4
     "@types/node": "*"
-  checksum: e921718775992ada7712a26da9635fb970bcdbcf4c4cc832344092d68710ffaf7dd3cce42b0050b2741b2cf75a30f3e1b87e1032ae26552db1faac17094d2b50
+  checksum: 885b7813676eed755b94d42501a56bf36c8d93d8ef6cc4d495bb5779fc5c81ffbcee7d0defe77515a90bf46209968da84b559471611efa0d2900ec75fa1a84de
   languageName: node
   linkType: hard
```

The changes in `yarn.lock` are the result of removing the existing entry for a package (`@types/vinyl`) that was triggering errors and re-running `yarn` to add the latest version.

Reply:
Done: https://github.com/hypothesis/h/pull/8789/commits/6c5ccec4c30d3d6600223e4c8c8500c89529e0fc